pmm/pmm2-api-tests.groovy

```diff
@@ -73,72 +69,88 @@ pipeline {
+                '''
                 script {
-                    if (!env.SERVER_IP) {
-                        runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1')
-                    }
-                    else
-                    {
-                        env.VM_IP = env.SERVER_IP
-                        env.PMM_URL = "http://admin:admin@${env.VM_IP}"
-                    }
+                    env.VM_IP = "127.0.0.1"
+                    env.PMM_URL = "http://admin:admin@${env.VM_IP}"
                 }
             }
         }
-        stage('Setup Step') {
-            parallel {
-                stage('Setup Docker')
-                {
-                    steps{
-                        sh '''
-                            sudo yum -y install docker
-                            sudo usermod -aG docker ec2-user
-                            sudo service docker start
-                            sudo docker build -t pmm-api-tests .
-                        '''
-                    }
-                }
-                stage('Sanity Check')
-                {
-                    steps {
-                        sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
-                    }
-                }
+        stage('Sanity Check')
+        {
+            steps {
+                sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
             }
         }
         stage('Run API Test') {
             steps {
                 sh '''
-                    sudo docker run -e PMM_SERVER_URL=\${PMM_URL} -e PMM_RUN_UPDATE_TEST=1 -e PMM_RUN_STT_TESTS=0 --name ${BUILD_TAG} pmm-api-tests
+                    docker run -e PMM_SERVER_URL=\${PMM_URL} -e PMM_RUN_UPDATE_TEST=1 -e PMM_RUN_STT_TESTS=0 --name ${BUILD_TAG} --network host pmm-api-tests
                 '''
             }
         }
     }
     post {
         always {
             sh '''
-                sudo docker ps -a
-                sudo docker cp ${BUILD_TAG}:/go/src/github.com/Percona-Lab/pmm-api-tests/pmm-api-tests-junit-report.xml ./${BUILD_TAG}.xml
-                ls -al
-                sudo docker stop ${BUILD_TAG}
-                sudo docker rm ${BUILD_TAG}
+                docker cp ${BUILD_TAG}:/go/src/github.com/Percona-Lab/pmm-api-tests/pmm-api-tests-junit-report.xml ./${BUILD_TAG}.xml || true
+                docker stop ${BUILD_TAG} || true
+                docker rm ${BUILD_TAG} || true
+                curl --insecure ${PMM_URL}/logs.zip --output logs.zip || true
+                cd pmm-agent
+                docker-compose down
+                docker rm -f $(sudo docker ps -a -q) || true
+                docker volume rm $(sudo docker volume ls -q) || true
+                cd ../
+                sudo chown -R ec2-user:ec2-user pmm-agent || true
```

Comment:
Still need this and the end of the pipeline?

Reply:
yes, delete plugin keeps failing due to permission, to avoid that I need to use this.